### ISSUE TITLE
Add note to docs about libtinfo dependency on linux

### ISSUE THREAD

**Issue by pchiusano** [unisonweb/unison#682]
If get:

Hi
Upon running `ucm` I'm getting
```
ucm: error while loading shared libraries: libtinfo.so.5: cannot open shared object file: No such file or directory
```
Do you know what package to install to fix this?
This is PopOS (Ubuntu)

sudo apt-get install libtinfo5

Fixes the issue.

**Comment by atacratic** [comment]
Data point: I didn't hit this on vanilla Ubuntu 18.04.2.

**Comment by fkohlgrueber** [comment]
I needed to manually install libinfo5 too (Ubuntu 19.04).

**Comment by pchiusano** [comment]
We are also thinking we should update home brew package to make this dependency explicit. (Linux users manually installing the binary would still need to make sure they have it installed so I’ll keep this ticket open.)

**Comment by BenFradet** [comment]
I have a variant of the above which do not seem to be getting in the way of running ucm, when launching it I get:

```bash
./ucm: /lib64/libtinfo.so.5: no version information available (required by ./ucm)
  ☝️
  
  No codebase exists here so I'm initializing one in: .unison/v1

   _____     _             
  |  |  |___|_|___ ___ ___ 
  |  |  |   | |_ -| . |   |
  |_____|_|_|_|___|___|_|_|
  
  Welcome to Unison!
  
  I'm currently watching for changes to .u files under /tmp/install
  
  Type help to get help. 😎

.>
```

I'm on fedora 30.

**Comment by aryairani** [comment]
Huh, that's interesting!

**Comment by aryairani** [comment]
I wonder if we just need to build our binary on a libtinfo 6 vm, and that that will more likely match what our users have?

**Comment by francisdb** [comment]
Ubuntu 18.04 (latest LTS) seems to com with libtinfo.so.5

```bash
> ll /lib/x86_64-linux-gnu/libtinfo*
lrwxrwxrwx 1 root root     15 May 23  2018 /lib/x86_64-linux-gnu/libtinfo.so.5 -> libtinfo.so.5.9
-rw-r--r-- 1 root root 170784 May 23  2018 /lib/x86_64-linux-gnu/libtinfo.so.5.9

> apt-cache search libtinfo
libtinfo-dev - developer's library for the low-level terminfo library
libtinfo5 - shared low-level terminfo library for terminal handling
libtinfo5-dbg - debugging/profiling library for the low-level terminfo library
```

**Comment by sfultong** [comment]
This issue is a great demonstration of why I prefer the nix build tool (see #622). You can pull in non-haskell build dependencies, no matter your distro (and it should probably work in OSX as well).

So... why aren't we just merging #622 and telling people to use nix, rather than trying to develop specific build instructions for every possible distro?

**Comment by aryairani** [comment]
@sfultong It's just that: 

a) we aren't familiar enough with nix at the office to manage keeping the nix definitions up to date with each subsequent PR & dependency (but also don't want them to be broken — the nix PR doesn't test the nix build in travis yet, and if we or some other contributor adds a new dependency that breaks the nix build, we aren't familiar enough to nix to be able to help fix it), 

b) my impression of nix was that it has a steep learning curve, so I'm not sure it imposing nix for builds would help any but the few existing nix/unison users, 

and c) because the irrelevant HIE stuff hasn't been removed from the PR yet.

My personal feeling is that it would make more sense at this stage for a nix guru to maintain a separate nixified fork, and pull in unisonweb/unison `master` at their leisure.  Change my mind?

**Comment by pchiusano** [comment]
Also, I don't think @sfultong is proposing this, but I wouldn't want to _require_ all our Linux users to install via Nix. Having vanilla Linux binaries is a nice thing IMO, even if there are a few weird cases like this.

I'm all for having as many convenient ways to install Unison as possible, including Nix, and if someone would like to maintain a Nix-ified fork and keep it up to date and cut releases, then that's great and we can add another link to the install guide.

**Comment by aryairani** [comment]
I'd like to update the brew definition, but basically don't know how.  I would also be open to sorting out .rpm / .deb packages, but not in an immediate position to help there either.

**Comment by aryairani** [comment]
https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops#use-a-microsoft-hosted-agent if we want to change the build container

**Comment by litriv** [comment]
> I have a variant of the above which do not seem to be getting in the way of running ucm, when launching it I get:
> 
> ```shell
> ./ucm: /lib64/libtinfo.so.5: no version information available (required by ./ucm)
>   ☝️
>   
>   No codebase exists here so I'm initializing one in: .unison/v1
> 
>    _____     _             
>   |  |  |___|_|___ ___ ___ 
>   |  |  |   | |_ -| . |   |
>   |_____|_|_|_|___|___|_|_|
>   
>   Welcome to Unison!
>   
>   I'm currently watching for changes to .u files under /tmp/install
>   
>   Type help to get help. 😎
> 
> .>
> ```
> 
> I'm on fedora 30.

Same here on FreeBSD running in Linux compatibility mode with Centos 7 libraries.  

My ucm also starts, but then when I install the base libraries and `cd .base` and then `ls`, there's nothing there, which is possibly an unrelated issue - I'll dig deeper and create a separate issue if necessary.

**Comment by kghenderson** [comment]
fwiw, i'm using Fedora Silverblue, and had to `sudo dnf install ncurses-compat-libs` within the toolbox to get this library
+1 for having native package installers (in addition to nix)

**Comment by hdgarrood** [comment]
Can this be closed and removed from the FAQ now that #2085 is in?

**Comment by ceedubs** [comment]
@hdgarrood thanks for the heads up. I believe you are correct. I'm going to close this out and @pchiusano you can open it back up if I'm mistaken.
